File: parse_figma.py
import urllib.request
import json

TOKEN = 'YOUR_FIGMA_TOKEN_HERE'
FILE_KEY = "avFrNcK17LQ5fX9LeeX94f"
NODE_ID = "1627:22833"

url = f"https://api.figma.com/v1/files/{FILE_KEY}/nodes?ids={NODE_ID}"
req = urllib.request.Request(url, headers={"X-Figma-Token": TOKEN})

try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode('utf-8'))
        nodes = data.get('nodes', {})
        node = nodes.get(NODE_ID, {}).get('document', {})
        
        def print_tree(n, indent=0):
            if not n: return
            prefix = "  " * indent
            name = n.get('name', 'Unnamed')
            t = n.get('type', 'UNKNOWN')
            chars = n.get('characters', '')
            
            fills = n.get('fills', [])
            color = ""
            if fills and isinstance(fills, list) and len(fills) > 0 and fills[0].get('type') == 'SOLID':
                c = fills[0].get('color', {})
                color = f" color=rgba({c.get('r',0)*255:.0f},{c.get('g',0)*255:.0f},{c.get('b',0)*255:.0f},{c.get('a',1):.2f})"
            
            text_str = f" text='{chars}'" if chars else ""
            print(f"{prefix}- [{t}] {name}{color}{text_str}")
            
            for child in n.get('children', []):
                print_tree(child, indent + 1)
                
        print_tree(node)
except Exception as e:
    print(f"Error: {e}")

File: parse_figma.dart
import 'dart:io';
import 'dart:convert';

void main() async {
  final token = 'YOUR_FIGMA_TOKEN_HERE';
  final fileKey = 'avFrNcK17LQ5fX9LeeX94f';
  final nodeId = '1627:22833';

  final url = Uri.parse('https://api.figma.com/v1/files/$fileKey/nodes?ids=$nodeId');
  
  final request = await HttpClient().getUrl(url);
  request.headers.add('X-Figma-Token', token);
  final response = await request.close();
  
  final responseBody = await response.transform(utf8.decoder).join();
  final data = json.decode(responseBody);
  
  final nodes = data['nodes'] as Map<String, dynamic>? ?? {};
  final node = (nodes[nodeId] as Map<String, dynamic>?)?['document'] as Map<String, dynamic>? ?? {};
  
  void printTree(Map<String, dynamic>? n, [int indent = 0]) {
    if (n == null) return;
    final prefix = '  ' * indent;
    final name = n['name'] ?? 'Unnamed';
    final t = n['type'] ?? 'UNKNOWN';
    final chars = n['characters'] ?? '';
    
    var color = '';
    final fills = n['fills'] as List<dynamic>?;
    if (fills != null && fills.isNotEmpty && fills[0]['type'] == 'SOLID') {
      final c = fills[0]['color'] as Map<String, dynamic>?;
      if (c != null) {
        final r = (c['r'] * 255).toInt();
        final g = (c['g'] * 255).toInt();
        final b = (c['b'] * 255).toInt();
        final a = c['a'];
        color = ' color=rgba($r,$g,$b,$a)';
      }
    }
    
    final textStr = chars.isNotEmpty ? ' text="$chars"' : '';
    print('$prefix- [$t] $name$color$textStr');
    
    final children = n['children'] as List<dynamic>?;
    if (children != null) {
      for (final child in children) {
        printTree(child as Map<String, dynamic>?, indent + 1);
      }
    }
  }
  
  printTree(node);
}
